src/components/ArticlesList.jsx

```diff
@@ -1,33 +1,54 @@
 import {useEffect, useState} from "react"
 import ArticleCard from "./ArticleCard"
 import getArticles from "../utils/getArticles"
-import { useParams } from "react-router-dom"
+import { useParams, useSearchParams } from "react-router-dom"
+import { Link } from "react-router-dom"
 
-function ArticlesList ({topics}) {  
+function ArticlesList () {  
     const [articles, setArticles] = useState([])
-    
     const { topic } = useParams()
 
+    const[searchParams, setSearchParams] = useSearchParams()
+    
+    const sortByQuery = searchParams.get("sort_by")
+    const orderQuery = searchParams.get("order")
+
+    const setSortOrder = (direction) => {
+        const newParams = new URLSearchParams(searchParams);
+        newParams.set('order', direction);
+        setSearchParams(newParams);
+      }
+
     useEffect(() => {
-        getArticles()
+        getArticles(sortByQuery, orderQuery)
         .then((response) => {
             setArticles(response)
         })
-    }, [])
+    }, [sortByQuery, orderQuery])
+
+    
+
 
     return ( 
-        <section>
-            {articles.map((article) => {
-                if(topic === article.topic){
-                    return <ArticleCard article={article} key={article.article_id}/>
-                }
-                else if (!topic){
-                    return <ArticleCard article={article} key={article.article_id}/>
-
-                }
-            })}
-            
-        </section>
+        <>
+        <Link to="?sort_by=created_at"><button >sort by date</button></Link>
```

Comment:
You could handle this without links, just manually setting the search params and refetching the data when you do